src/gig_argument.c

```diff
@@ -673,6 +678,71 @@ scm_to_c_ptr_array(S2C_ARG_DECL)
     free(_arg.v_pointer);
 }
 
+static void
+scm_to_c_ghashtable(S2C_ARG_DECL)
+{
+    TRACE_S2C();
+    GHashFunc hash_func;
+    GEqualFunc equal_func;
+
+    // To make GHashTable from this SCM, we need to guess its
+    // hash function and equality function.
+    if (!meta->params[0].is_ptr) {
+        hash_func = g_direct_hash;
+        equal_func = g_direct_equal;
+    }
+    else if (meta->params[0].gtype == G_TYPE_INT) {
+        hash_func = g_int_hash;
+        equal_func = g_int_equal;
+    }
+    else if (meta->params[0].gtype == G_TYPE_INT64) {
+        hash_func = g_int64_hash;
+        equal_func = g_int64_equal;
+    }
+    else if (meta->params[0].gtype == G_TYPE_DOUBLE) {
+        hash_func = g_double_hash;
+        equal_func = g_double_equal;
+    }
+    else if (meta->params[0].gtype == G_TYPE_STRING) {
+        hash_func = g_str_hash;
+        equal_func = g_str_equal;
+    }
+    else {
+        hash_func = NULL;
+        equal_func = NULL;
+    }
```

Comment:
That seems to be a good enough guess for now, but we should probably give the user power over this part. (See my rant on object properties).

Reply:
I like the idea, but, I just wanted to commit this minimum viable patch before things got more complex.